server/sonar-web/src/main/js/apps/sessions/components/LoginForm.js

```diff
@@ -22,26 +22,49 @@ import React from 'react';
 import GlobalMessagesContainer from '../../../app/components/GlobalMessagesContainer';
 import { translate } from '../../../helpers/l10n';
 
+type Props = {
+  identityProviders: Array<{
+    backgroundColor: string,
+    iconPath: string,
+    key: string,
+    name: string
+  }>,
+  onSubmit: (string, string) => void
+};
+
+type State = {
+  collapsed: boolean,
+  login: string,
+  password: string
+};
+
 export default class LoginForm extends React.PureComponent {
-  static propTypes = {
-    identityProviders: React.PropTypes.array.isRequired,
-    onSubmit: React.PropTypes.func.isRequired
-  };
+  props: Props;
+  state: State;
 
-  state = {
-    login: '',
-    password: ''
-  };
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      collapsed: props.identityProviders.length > 0,
+      login: '',
+      password: ''
+    };
+  }
 
-  handleSubmit = (e: Object) => {
-    e.preventDefault();
+  handleSubmit = (event: Event) => {
+    event.preventDefault();
     this.props.onSubmit(this.state.login, this.state.password);
   };
 
+  handleMoreOptionsClick = (event: Event) => {
+    event.preventDefault();
+    this.setState({ collapsed: false });
+  };
+
   render() {
     return (
-      <div>
-        <h1 className="maintenance-title text-center">Log In to SonarQube</h1>
+      <div id="login_form">
+        <h1 className="maintenance-title text-center">{translate('login.login_to_sonarqube')}</h1>
```

Comment:
the request to settings api to known if the current instance is sonarcloud is not done for the login page (and it's not even possible if the forced authentication is enabled)